posthog/api/capture.py

```diff
@@ -359,40 +359,48 @@ def drop_events_over_quota(token: str, events: list[Any]) -> EventsOverQuotaResu
     if not settings.EE_AVAILABLE:
         return EventsOverQuotaResult(events, False, False)
 
-    from ee.billing.quota_limiting import QuotaResource, list_limited_team_attributes
-
-    results = []
+    # Get lists of limited tokens from Redis
     limited_tokens_events = list_limited_team_attributes(
         QuotaResource.EVENTS, QuotaLimitingCaches.QUOTA_LIMITER_CACHE_KEY
     )
     limited_tokens_recordings = list_limited_team_attributes(
         QuotaResource.RECORDINGS, QuotaLimitingCaches.QUOTA_LIMITER_CACHE_KEY
     )
 
+    results = []
     recordings_were_limited = False
     events_were_limited = False
+
     for event in events:
+        # Check if it's a recording event
         if event.get("event") in SESSION_RECORDING_EVENT_NAMES:
             EVENTS_RECEIVED_COUNTER.labels(resource_type="recordings").inc()
             if token in limited_tokens_recordings:
                 EVENTS_DROPPED_OVER_QUOTA_COUNTER.labels(resource_type="recordings", token=token).inc()
                 if settings.QUOTA_LIMITING_ENABLED:
                     recordings_were_limited = True
                     continue
-
         else:
+            # Regular events - check if they're free (i.e. non-quota limited) events first
+            event_name = event.get("event")
+            is_free_event = (
+                event_name in ("$exception", "survey sent", "survey shown", "survey dismissed", "$feature_flag_called")
+                or event_name.startswith("$ai_")  # AI events, right now we don't charge for these
+                or event_name.startswith(
+                    "$error_"
+                )  # Error tracking events, same with AI, but we're start charging once it's out of beta
+            )
+
             EVENTS_RECEIVED_COUNTER.labels(resource_type="events").inc()
-            if token in limited_tokens_events:
+            if token in limited_tokens_events and not is_free_event:
```

Comment:
@oliverb123 do I need to do something similar for the Rust capture service?  Not really sure how that service does quota limiting but I imagine you have a comparable implementation, just not sure exactly where to look for it.